dspace-server-webapp/src/main/java/org/dspace/app/rest/signposting/processor/bitstream/BitstreamTypeProcessor.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.signposting.processor.bitstream;
+
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+import org.dspace.app.rest.signposting.model.LinksetNode;
+import org.dspace.app.rest.signposting.model.LinksetRelationType;
+import org.dspace.content.Bitstream;
+import org.dspace.content.Item;
+import org.dspace.content.service.BitstreamService;
+import org.dspace.core.Context;
+import org.dspace.util.FrontendUrlService;
+import org.dspace.util.SimpleMapConverter;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * An extension of {@link BitstreamSignpostingProcessor} for the type relation.
+ * Provides links to a specific type from schema.org.
+ */
+public class BitstreamTypeProcessor extends BitstreamSignpostingProcessor {
+
+    private static final Logger log = Logger.getLogger(BitstreamTypeProcessor.class);
+
+    @Autowired
+    private SimpleMapConverter mapConverterDSpaceToSchemaOrgUri;
+
+    @Autowired
+    private BitstreamService bitstreamService;
+
+    public BitstreamTypeProcessor(FrontendUrlService frontendUrlService) {
+        super(frontendUrlService);
+        setRelation(LinksetRelationType.TYPE);
+    }
+
+    @Override
+    public void addLinkSetNodes(Context context, HttpServletRequest request,
+                                Bitstream bitstream, List<LinksetNode> linksetNodes) {
+        try {
+            String type = bitstreamService.getMetadataFirstValue(bitstream, "dc", "type", null, Item.ANY);
```

Comment:
I'm confused by this Processor because Bitstreams don't normally have a `dc.type`.  That field is on Items.  Should this be looking for the Bitstream Mime Type instead?  If we are already returning the MIME type elsewhere, then maybe we don't need this?

Reply:
You are right Bitstreams normally don't have a `dc.type`, but if it provided anyway it will be processed by that processor. It also can be disabled in `signposting.xml`